### DISCUSSION TITLE
txt2img - hires only - using image that generated in low res.

### DISCUSSION THREAD

**Discussion by shtorm2005** [Mikubill/sd-webui-controlnet#2828] [Ideas]
When activating Hires.fix in txt2img tab, hires option becomes available in controlnet that allows you to use controlnet during lowres/hires step, but you need to provide a separate image for this to work. 

It would be nice to have a feature where for hires only, controlnet uses an image created during lowres step. 

One of advantages, I see, is for openpose controlnet. For example, generated lowres image has bad hands, in hires step, it will be fixed using controlnet openpose.

Sorry for my english.

**Comment by huchenlei** [comment]
Are you looking for after detailing? https://github.com/Bing-su/adetailer

**Reply by shtorm2005** [reply]
Adetailer is great, but there is an advantage to control your image during hires fix and not trying to fix after image is generated. Fixing with Adetailer a big image takes high denoise which makes inpaint area less appropriate to the whole image. I don't know if you understand what I mean.

**Reply by huchenlei** [reply]
Currently you can already do it, see https://github.com/Mikubill/sd-webui-controlnet/pull/2379:
- Generate an image
- Fix the seed and other params
- Use the generated image as input in controlnet and select hi-res pass only for HrOption
- Generate with HR Fix enabled

**Reply by shtorm2005** [reply]
Yes, you can just drag and drop a generated image to controlnet each time, but it would be great to make it automatic because you often work with batches in txt2img. (to find good ones in many generated)

As for now, you can **automatically** apply controlnet to hires.fix using different images, but not the original lowres image.

It's just an idea. I think something similar is already implemented in img2img tab, 

![изображение](https://github.com/Mikubill/sd-webui-controlnet/assets/148022087/2c904f0e-4fa8-458d-b046-65ac736ea246)

**Reply by huchenlei** [reply]
I am not familar with the batch logic currently implemented, but the general guideline is if you want something fancier than using a list of images as ControlNet inputs, you should use [API](https://github.com/Mikubill/sd-webui-controlnet/wiki/API) to achieve these custom logic.

**Reply by huchenlei** [reply]
PRs are welcomed to refactor existing batch logic, as I personally do not use batch.

Workflow sharing dicussion is also welcomed.
